### PR TITLE
Add CustomButton component and replace Button with CustomButton

### PR DESCRIPTION
**Pull request by alexchen45**
## Why do we need this PR?
* Button component is replaced with new CustomButton Component
* Color and shadows of buttons are altered.
## How did you address the issue?
*

### CODE REVIEW

**Review by jiangxin1998 — APPROVED** [review]

**Review by KalleChen — CHANGES_REQUESTED** [review]
Look great, but there is still something to be changed. And we should now follow the rule of eslint, please check eslint every time you create a PR.

File: src/utils/functions/ImageUpload.js [hunk 1]
```diff
@@ -30,17 +28,14 @@ function ImageUpload(props) {
...
         onClick={fileButtonClick}
-        style={{
-          filter: 'drop-shadow(0px 4px 4px rgba(0, 0, 0, 0.25))',
-          borderRadius: '20px'
-        }}
-      >
-        選擇照片
-      </Button>
+        borderRadius='20px'
+        children='選擇照片'
```
Comment (KalleChen) [inline src/utils/functions/ImageUpload.js:37]: eslint

File: src/pages/MapPage/components/Mission/MissionDrawer/MissionStepper.js [hunk 2]
```diff
@@ -97,29 +95,30 @@ function MissionStepper(props) {
...
-                    : () => setRemindOpen(true)
-                }
-                className={classes.button}
-              >
-                確定
-              </Button>
+                  ? () => setFinishOpen(true)
+                  : () => setRemindOpen(true)
+                  }
+                children='確定'
```
Comment (KalleChen) [inline src/pages/MapPage/components/Mission/MissionDrawer/MissionStepper.js:107]: eslint

File: src/pages/MapPage/components/Mission/MissionDrawer/MissionStepper.js [hunk 3]
```diff
@@ -68,22 +66,22 @@ function MissionStepper(props) {
...
-            <Button
+            <CustomButton
+              className={classes.button}
               size='medium'
               onClick={handleCompleteStreetView}
-              className={classes.button}
-            >
-              確定
-            </Button>
+              children='確定'
```
Comment (KalleChen) [inline src/pages/MapPage/components/Mission/MissionDrawer/MissionStepper.js:82]: eslint

File: src/pages/MapPage/components/Mission/MissionDrawer/MissionStep2/MissionStep2.js [hunk 4]
```diff
@@ -172,19 +173,18 @@ function MissionStep2() {
...
                   ? 'primary'
                   : 'default'
               }
-              onClick={() => handleSetSelectedMissionId(facility.subTypeName)}
-            >
-              {facility.subTypeName}
-            </Button>
+              size='small'
+              fullWidth
+              children={facility.subTypeName}
```
Comment (KalleChen) [inline src/pages/MapPage/components/Mission/MissionDrawer/MissionStep2/MissionStep2.js:186]: eslint

File: src/pages/MapPage/components/Mission/MissionDrawer/MissionStep1/MissionStep1.js [hunk 5]
```diff
@@ -69,19 +68,29 @@ function MissionStep1() {
...
-        </Button>
+        </Button> */}
+        <CustomButton
+          variant='contained'
+          onClick={() => {
+            setStep(MissionStep.PlaceFlagOnStreet)
+          }}
+          color={streetViewUpload ? 'primary' : 'default'}
+          children='新增街景'
+          borderRadius='20px'
```
Comment (KalleChen) [inline src/pages/MapPage/components/Mission/MissionDrawer/MissionStep1/MissionStep1.js:91]: border-radius

File: src/pages/MapPage/components/Filter/FilterDrawer.js [hunk 6]
```diff
@@ -144,34 +139,34 @@ const FilterDrawer = (props) => {
...
           color={final ? 'primary' : 'default'}
-          disabled={!final}
           variant='contained'
-        >
-          加入
-        </Button>
+          onClick={handleFinishFilter}
+          disabled={!final}
+          children='加入'
+          borderRadius='20px'
```
Comment (KalleChen) [inline src/pages/MapPage/components/Filter/FilterDrawer.js:168]: border-radius should be write in the CustomButton file. If we have two version of border-radius, create a prop to choose which to use, we should not passing the style value here.

File: src/pages/MapPage/components/Filter/FilterDrawer.js [hunk 7]
```diff
@@ -144,34 +139,34 @@ const FilterDrawer = (props) => {
...
                           ? 'default'
                           : 'primary'
                       }
+                      variant='contained'
+                      size='small'
                       onClick={() => changeCurrentTarget(discovery.targetName)}
-                    >
-                      {discovery.targetName}
-                    </Button>
+                      children={discovery.targetName}
```
Comment (KalleChen) [inline src/pages/MapPage/components/Filter/FilterDrawer.js:152]: eslint

File: src/pages/MapPage/components/Filter/FilterDrawer.js [hunk 8]
```diff
@@ -122,17 +118,16 @@ const FilterDrawer = (props) => {
...
                         currentSubmission !== discovery ? 'default' : 'primary'
                       }
-                      onClick={() => changeCurrentSubmission(discovery)}
-                    >
-                      {discovery.subTypeName}
-                    </Button>
+                      variant='contained'
+                      size='small'
+                      onClick={()=> changeCurrentSubmission(discovery)}
+                      children={discovery.subTypeName}
```
Comment (KalleChen) [inline src/pages/MapPage/components/Filter/FilterDrawer.js:129]: eslint

File: src/pages/MapPage/components/Filter/FilterDrawer.js [hunk 9]
```diff
@@ -102,17 +101,14 @@ const FilterDrawer = (props) => {
...
-                  onClick={() => {
-                    changeCurrentMission(mission)
-                  }}
-                >
-                  {mission.missionName}
-                </Button>
+                  variant='contained'
+                  size='small'
+                  onClick={()=> changeCurrentMission(mission)}
+                  children={mission.missionName}
```
Comment (KalleChen) [inline src/pages/MapPage/components/Filter/FilterDrawer.js:110]: elsint

File: src/components/CustomDrawer/CustomDrawer.js [hunk 10]
```diff
@@ -106,14 +105,13 @@ const CustomDrawer = (props) => {
...
-                <Button
+                <CustomButton
                   className={classes.titleActionButton}
                   size='small'
                   onClick={titleAction.handleOnClick}
                   disabled={titleAction.disable}
-                >
-                  {titleAction.name}
-                </Button>
+                  children={titleAction.name}
```
Comment (KalleChen) [inline src/components/CustomDrawer/CustomDrawer.js:113]: For eslint rule, do not pass children from props

File: src/components/CustomDrawer/CustomDrawer.js [hunk 11]
```diff
@@ -50,8 +50,7 @@ const useStyles = makeStyles((theme) => ({
   },
   titleActionButton: {
     background: '#D8D8D8',
-    border: '1px solid #BABABA',
-    boxShadow: '0px 3px 4px rgba(0, 0, 0, 0.12)',
+    boxShadow: '0px 2px 4px rgba(0, 0, 0, 0.12)',
     borderRadius: '20px'
   }
```
Comment (KalleChen) [inline src/components/CustomDrawer/CustomDrawer.js:55]: We should remove this here, the main goal of this feature is that we can only import the button and use it without create other styles.

File: src/components/CustomButton/CustomButton.js [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+            color: color==='primary' ?'#000000'
+                :color==='default'?'#686868'
+                :{color},
+            backgroundColor: color==='primary' ?'#FDCC4F'
+                :color==='default'?'#EEEEEE'
+                :{backgroundColor},
+            boxShadow: noShadow==='true'
+                ?'0px 0px 0px rgba(0, 0, 0, 0)'
+                :'0px 2px 2px rgba(0, 0, 0, 0.12)',
+            borderRadius: borderRadius,
```
Comment (KalleChen) [inline src/components/CustomButton/CustomButton.js:27]: And the default style should be implemented with material ui's `withStyle` or `makeStyles`, you can check how CustomerDrawer does.

File: src/components/CustomButton/CustomButton.js [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+          variant={variant}
+          size={size}
+          //color={color}
+          style={{
+            color: color==='primary' ?'#000000'
+                :color==='default'?'#686868'
+                :{color},
+            backgroundColor: color==='primary' ?'#FDCC4F'
+                :color==='default'?'#EEEEEE'
+                :{backgroundColor},
```
Comment (KalleChen) [inline src/components/CustomButton/CustomButton.js:23]: color and backgroundColor should be implemented in color prop `color={color}` and the color value is setting in the `theme.js`.

**Review by KalleChen — CHANGES_REQUESTED** [review]
Look great, but there is still something to be changed. And we should now follow the rule of eslint, please check eslint every time you create a PR.

**Review by KalleChen — APPROVED** [review]

**Pull request merged**